litestar/cli/_utils.py

```diff
@@ -410,3 +423,127 @@ def show_app_info(app: Litestar) -> None:  # pragma: no cover
         table.add_row("Middlewares", ", ".join(middlewares))
 
     console.print(table)
+
+
+def _validate_file_path(file_path: str | None) -> Path | None:
+    """Validate whether a path was provided, exists and is not a directory. Return the resolved path."""
+    if file_path is None:
+        return None
+
+    path = Path(file_path).resolve()
+
+    if path.is_dir():
+        raise LitestarCLIException(f"Provided path is a directory: {path}")
+
+    if not path.exists():
+        return None
+
+    return path
+
+
+def _validate_new_file_path(file_path: str) -> Path:
+    """Validate if a path's parent directory exist. Return the resolved path."""
+    path = Path(file_path).resolve()
+
+    if not (parent_dir := path.parent).exists():
+        raise LitestarCLIException(f"Unable to create file. Directory doesn't exist: {parent_dir}")
+
+    return path
+
+
+def validate_and_create_ssl_files(
```

Comment:
This should be split into two different functions. You have to branches here, guarded by the `create_devcert` bool, that have completely different behaviour.

Additionally, I think the logic can be simplified a bunch. We don't need to do the whole path validation thing if we're not creating a devcert. Uvicorn will validate the file paths provided anyway. So all we need to do is, in case the `create-devcert` flag is passed, check if both paths provided do not exist, in which case we generate a cert, or if only one of them exists, in which case we raise an exception.

Reply:
@provinzkraut
> You have to branches here

The problem is after adding a single `if` to the function, it doesn't pass the pre-commit checks, ruff complains that it is too complex ([C901](https://www.flake8rules.com/rules/C901.html))

> I think the logic can be simplified a bunch. We don't need to do the whole path validation thing if we're not creating a devcert. Uvicorn will validate the file paths provided anyway

The uvicorn validation is very bare bones and the error messages are not nice and not even consistent for when the key file is missing vs cert file is missing. Given that it might be not intuitive what different combinations of flags do, I wanted to give nice and explicit CLI errors that would guide the user. 
Do you really want to get rid of that?

Reply:
> The problem is after adding a single `if` to the function, it doesn't pass the pre-commit checks, ruff complains that it is too complex ([C901](https://www.flake8rules.com/rules/C901.html))

That's why I'm suggesting to split this into 2 functions. This is a classic "boolean trap"; A function that has two completely different behaviours that depend on a boolean argument passed to the function. The only thing these branches have in common is that they return the same thing. This complexity can be avoided by taking what's currently in the branches, putting it into two separate functions, and calling them based on the boolean you'd pass to the flag. 


> The uvicorn validation is very bare bones and the error messages are not nice and not even consistent for when the key file is missing vs cert file is missing.

Fair point. If we're adding value here then it's worth having this validation :+1:

Reply:
I'll deal with the linter and try to simplify it

Reply:
@provinzkraut I understand that and hopefully have fixed it. The complexity warning was on the already existing `run_command` function so I silenced it there. My code should be a cleaner now